processor/telemetryprocessor/telemetry_processor_test.go

```diff
@@ -0,0 +1,216 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package telemetryprocessor
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.opencensus.io/stats/view"
+	"go.uber.org/zap"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/component/componenttest"
+	"go.opentelemetry.io/collector/config/configtelemetry"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.opentelemetry.io/collector/internal/collector/telemetry"
+	"go.opentelemetry.io/collector/internal/data/testdata"
+	"go.opentelemetry.io/collector/obsreport/obsreporttest"
+	"go.opentelemetry.io/collector/processor"
+	"go.opentelemetry.io/collector/translator/conventions"
+)
+
+func TestErrorPath(t *testing.T) {
+	doneFn, err := obsreporttest.SetupRecordedMetricsTest()
+	require.NoError(t, err)
+	defer doneFn()
+
+	views := processor.MetricViews(configtelemetry.LevelDetailed)
+	assert.NoError(t, view.Register(views...))
+	defer view.Unregister(views...)
+
+	mockP := newMockSpanProcessor()
+	creationParams := component.ProcessorCreateParams{Logger: zap.NewNop()}
+	cfg := createDefaultConfig().(*Config)
+	qp := newTelemetryTracesProcessor(creationParams, mockP, cfg)
+	require.NoError(t, qp.Start(context.Background(), componenttest.NewNopHost()))
+	t.Cleanup(func() {
+		assert.NoError(t, qp.Shutdown(context.Background()))
+	})
+
+	mockP.consumeError = errors.New("consume error")
+
+	wantBatches := 10
+	for i := 0; i < wantBatches; i++ {
+		td := testdata.GenerateTraceDataTwoSpansSameResource()
+		require.Error(t, qp.ConsumeTraces(context.Background(), td))
+	}
+
+	obsreporttest.CheckProcessorTracesViews(t, cfg.Name(), int64(0), 20, 0)
+}
+
+func TestTraceHappyPath(t *testing.T) {
+	doneFn, err := obsreporttest.SetupRecordedMetricsTest()
+	require.NoError(t, err)
+	defer doneFn()
+
+	lvl, err := telemetry.GetLevel()
+	require.NoError(t, err)
+	telemetry.SetLevel(configtelemetry.LevelDetailed)
+	defer telemetry.SetLevel(lvl)
+
+	views := processor.MetricViews(configtelemetry.LevelDetailed)
+	assert.NoError(t, view.Register(views...))
+	defer view.Unregister(views...)
+
+	mockP := newMockSpanProcessor()
+	creationParams := component.ProcessorCreateParams{Logger: zap.NewNop()}
+	cfg := createDefaultConfig().(*Config)
+	qp := newTelemetryTracesProcessor(creationParams, mockP, cfg)
+	require.NoError(t, qp.Start(context.Background(), componenttest.NewNopHost()))
+	t.Cleanup(func() {
+		assert.NoError(t, qp.Shutdown(context.Background()))
+	})
+
+	wantBatches := 10
+	wantSpans := 20
+	for i := 0; i < wantBatches; i++ {
+		td := testdata.GenerateTraceDataTwoSpansSameResource()
+		rs := td.ResourceSpans()
+		for j := 0; j < rs.Len(); j++ {
+			r := rs.At(j)
+			r.Resource().Attributes().InsertString(conventions.AttributeServiceName, "foobar")
```

Comment:
is there a better way of doing this?  I don't like relying on OC conventions but pulling in otel/semconv seems large.